crates/ark/src/lsp/main_loop.rs

```diff
@@ -400,17 +404,18 @@ impl AuxiliaryState {
         // Channels for communication with the auxiliary loop
         let (auxiliary_event_tx, auxiliary_event_rx) = tokio_unbounded_channel::<AuxiliaryEvent>();
 
-        // Set global instance of this channel. This is used for interacting
-        // with the auxiliary loop (logging messages or spawning a task) from
-        // free functions.
-        unsafe {
-            if let Some(val) = AUXILIARY_EVENT_TX.get_mut() {
-                // Reset channel if already set. Happens e.g. on reconnection after a refresh.
-                *val = auxiliary_event_tx;
-            } else {
-                // Set channel for the first time
-                AUXILIARY_EVENT_TX.set(auxiliary_event_tx).unwrap();
-            }
+        // Set global instance of this channel. This is used for interacting with the
+        // auxiliary loop (logging messages or spawning a task) from free functions.
+        // Unfortunately this can theoretically be reset at any time, i.e. on reconnection
+        // after a refresh, which is why we need an RwLock. This is the only place we take
+        // a write lock though.
+        match AUXILIARY_EVENT_TX.write() {
+            Ok(mut tx) => {
+                *tx = Some(auxiliary_event_tx);
+            },
+            Err(error) => {
+                log::error!("Failed to set auxiliary event channel. {error}");
```

Comment:
I'd unwrap that since it means the auxiliary loop thread panicked, i.e. something very wrong happened?